### PR TITLE
Fixed a dropping asset issue after changing selected photos on the system setting.

### PR DESCRIPTION
**Pull request by Noodlekim**
**Fix note**
- Fixed a sync asset issue after changing selected photos on the system setting. For example, it has caused a sync issue when a user is ready to upload the selected photos on the system setting uploading photos.

### CODE REVIEW

File: NohanaImagePicker/AssetListSelectableDateSectionController.swift [hunk 1]
```diff
@@ -378,6 +378,30 @@ extension AssetListSelectableDateSectionController: PHPhotoLibraryChangeObserver
...
+                    let asset = afterResult.object(at: index)
+                    afterAssets.append(asset)
+                }
+
+                for index in 0..<beforeResult.count {
+                    let asset = beforeResult.object(at: index)
+                    if !afterAssets.map({ $0.localIdentifier }).contains(asset.localIdentifier) {
+                        removedAssets.append(asset)
+                    }
+                }
```
Comment (tyokujin) [inline NohanaImagePicker/AssetListSelectableDateSectionController.swift:398]: This code didn't delete the picked photo.

File: NohanaImagePicker/AssetListSelectableDateSectionController.swift [hunk 2]
```diff
@@ -378,6 +378,30 @@ extension AssetListSelectableDateSectionController: PHPhotoLibraryChangeObserver
...
+
+                for removedAsset in removedAssets {
+                    self.nohanaImagePickerController.dropAsset(PhotoKitAsset(asset: removedAsset))
+                }
+            }
```
Comment (tyokujin) [inline NohanaImagePicker/AssetListSelectableDateSectionController.swift:403]: We recommend syncing in the following ways.

Please refer 
```swift
            var removedAssets: [PhotoKitAsset] = []
            self.nohanaImagePickerController.pickedAssetList.forEach { item in
                if let asset = item as? PhotoKitAsset, let changeDetail = changeInstance.changeDetails(for: asset.asset) {
                    if changeDetail.objectWasDeleted {
                        removedAssets.append(asset)
                    }
                }
            }
            removedAssets.forEach { asset in
                self.nohanaImagePickerController.dropAsset(asset)
            }
            if !removedAssets.isEmpty {
                self.setToolbarTitle(self.nohanaImagePickerController)
            }
```

Comment (Noodlekim) [reply NohanaImagePicker/AssetListSelectableDateSectionController.swift:392]: I've modified this logic following your recommendation.
That seems to be better than before, moreover simple.

File: NohanaImagePicker/PhotoKitAssetList.swift [hunk 3]
```diff
@@ -74,4 +74,9 @@ open class PhotoKitAssetList: ItemList {
     open var endIndex: Int {
         return fetchResult.count
     }
+
+    open func changeDetails(_ changeInstance: PHChange) -> PHFetchResultChangeDetails<PHAsset>? {
```
Comment (tyokujin) [inline NohanaImagePicker/PhotoKitAssetList.swift:78]: Please delete unused function.

File: NohanaImagePicker/AssetListSelectableDateSectionController.swift (hunk 2 above)
Comment (tyokujin) [reply NohanaImagePicker/AssetListSelectableDateSectionController.swift:392]: Is toolbar title string update?
if it's not update, please run `setToolbarTitle()` function.

**Review by tyokujin — APPROVED** [review]

**Pull request merged**